pennylane/math/fn.py

```diff
@@ -323,13 +326,17 @@ def circuit(weights):
         o1 = obs[i]
         o2 = obs[j]
 
+        o1wires = o1.wires.labels if wires is None else wires.indices(o1.wires)
+        o2wires = o2.wires.labels if wires is None else wires.indices(o2.wires)
+        shared_wires = o1wires + o2wires

```

Comment:
Wait, the sum would be combined wires, not shared ones, right? If you want to make sure here that the obs do not act on the same wires you could do `Wires.all_wires(o1.wires, o2.wires)` and then extract label or indices?

Reply:
Ah! The reason I did this was a bit weird. Basically, by the time this variable ends up in low-level processing functions, the duplicated wires no longer matter. So I removed the call to `all_wires`, because of overhead fear 😆 

Including it is definitely more 'clean', but it adds unnecessary overhead...

How about this, as a cleaner way of writing it?

```suggestion
        shared_wires = set(o1wires + o2wires)
```